projecttacoma/deqm-test-server-frontend
# Resource id pagination

## Summary
Implementation of a Pagination component for browsing resource IDs of resourceTypes with more than 10 resources. 
## New behavior
On any resourceType home page where the resourceType has >10 resources, a Pagination component is rendered for navigation through all the resource IDs. 
  - The url is updated to include `?page=n` when a user navigates to any page number n.
  - If a resourceType has 10 or fewer resources, no Pagination component is rendered.
  - If a user attempts to navigate to an invalid page number (page num exceeds number of actual pages for the resourceType), then a "No resources found" message is displayed.
## Code changes
- **pages/[resource]/index.tsx**  renders pagination component 
- **__tests__/pages/resource/index.test.tsx**, unit tests added to file
- **components/ResourceIDs.tsx**, minor UI change
- **pages/[resourceType]/[id]/care-gaps.tsx**, **pages/[resourceType]/[id]/evaluate.tsx**, documentation comments updated
## Testing Guidance
- See the README.md for first time setup instructions.
#### Testing in browser: 
To start the frontend: `npm run dev` then navigate to http://localhost:3001 in a browser
- Click on a resourceType in the left hand side navigation menu that has more than 10 resources. At the home page for that resourceType, there will be 10 resource IDs rendered, along with a pagination component below the IDs.
  - Click on any numbered button in the pagination component to render that page’s resource ID list.
- Switch to a resourceType with 10 or fewer resources. There should be no pagination component rendered with the resource IDs.
- Delete a resource from a resourceType that has 11 resources to see the pagination component rendered for 11, then not rendered for 10.
- Navigate to a url with no ?page=n (i.e. http://localhost:3001/ValueSet). Page 1 of the resource IDs should be rendered.
- Navigate to a url with a ?page=n where n exceeds the number of pages for that resource. A "No resources found" message should be displayed.
#### Unit testing: 
- Run the unit tests: `npm run test`
